addons/point_of_sale/models/pos_config.py

```diff
@@ -525,9 +522,23 @@ def assign_payment_journals(self, companies=False):
         for company in companies:
             if company.chart_template_id:
                 cash_journal = self.env['account.journal'].search([('company_id', '=', company.id), ('type', '=', 'cash')], limit=1)
-                cash_journal.write({'journal_user':True})
-                existing_pos_config = self.env['pos.config'].search([('company_id', '=', company.id), ('journal_ids', '=', False)])
-                existing_pos_config.write({'journal_ids': [(6, 0, cash_journal.ids)]})
+                pos_receivable_account = company.default_pos_receivable_account_id
+                payment_methods = self.env['pos.payment.method']
+                payment_methods |= payment_methods.create({
```

Comment:
if cash_journal, if no cash journal, don't create payment method. (I know that it should not happend that there are no cash journal when we call this function, but it's more sure.